### DISCUSSION TITLE
Alternatives to BarCodeScanner.scanFromURLAsync() after BarCodeScanner deprecation

### DISCUSSION THREAD

**Discussion by chungcf2006** [expo/expo#26848] [Proposals and Ideas]
As stated in the documentation, BarCodeScanner will be deprecated from SDK 51. However, the recommended alternatives `expo-camera/next` does not support `BarCodeScanner.scanFromURLAsync()`, which allows an image to be imported and extract barcode data from it.

Does it mean that another third-party library has to be used or this should also be migrated to `expo-camera/next`?

**Comment by jmbmunda** [comment]
+1

**Reply by MoukeyCoin** [reply]
+1

**Reply by dShot7211** [reply]
I'm also looking for a solution.....

**Reply by webabbey247** [reply]
I'm currently using expo camera and  you need to replace BarCodeScanner.scanFromURLAsync()  with Camera.scanFromURLAsync(). Don't forget to import Camera from expo camera.

**Reply by gabriel-logan** [reply]
> I'm currently using expo camera and you need to replace BarCodeScanner.scanFromURLAsync() with Camera.scanFromURLAsync(). Don't forget to import Camera from expo camera.

This work
